src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20230810_150000.java

```diff
@@ -0,0 +1,86 @@
+package de.tum.in.www1.artemis.config.migration.entries;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import com.google.common.collect.Lists;
+
+import de.tum.in.www1.artemis.config.migration.MigrationEntry;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.security.SecurityUtils;
+import de.tum.in.www1.artemis.service.hestia.ProgrammingExerciseTaskService;
+
+/**
+ * Migration for automatically updating all Programming exercise problem statements, removing their test names and replacing them by test ids.
+ */
+public class MigrationEntry20230810_150000 extends MigrationEntry {
+
+    private final Logger log = LoggerFactory.getLogger(MigrationEntry20230810_150000.class);
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingExerciseTaskService taskService;
+
+    private static final int BATCH_SIZE = 100;
+
+    private static final int THREADS = 10;
+
+    public MigrationEntry20230810_150000(ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingExerciseTaskService taskService) {
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.taskService = taskService;
+    }
+
+    @Override
+    public void execute() {
+        long size = programmingExerciseRepository.count();
+        log.info("Migrating all {} programming exercises. This might take a while.", size);
+
+        ExecutorService executorService = Executors.newFixedThreadPool(THREADS);
+        Pageable pageable = Pageable.ofSize(BATCH_SIZE);
+
+        while (true) {
```

Comment:
You used an indexed loop in your migration, with such an approach this would be possible since you know how many pages you expect. I use the check for the last page in my code, and I only know information about the current page inside of the loop.

Reply:
Whoopsie. Well ^^ 
It's the migration anyway, so not a big issue.